### PR TITLE
Annotate - highlight recent file revisions lines

### PR DESCRIPTION
**Pull request by dobrou**
On annotate page - color file revisions numbers background, according to their age, so that recent revisions are highlighted more.

This way it's easy to visually distinguish new/old parts of file, without exactly reading revision ids.

Preview here:
opengrok-annotate-highlight

I will welcome any feedback, especially for css part. This could cooperate better with regular css styles in OpenGrok.

### CODE REVIEW

File: src/org/opensolaris/opengrok/history/HistoryGuru.java [hunk 1]
```diff
@@ -154,6 +155,8 @@ public Annotation annotate(File file, String rev) throws IOException {
                     ret.addDesc(short_rev, "changeset: " + he.getRevision() +
                         "\nsummary: " + he.getMessage() + "\nuser: " +
                         he.getAuthor() + "\ndate: " + he.getDate());
+                    ret.addFileVersion(short_rev, revs.size() - revsMatched); //history entries are coming from recent to older, file version should be from oldest to newer
```
Comment (vladak) [inline src/org/opensolaris/opengrok/history/HistoryGuru.java]: pls keep the lines short

**Comment by vladak** [comment]
Couple of comments:
  - the color should not be hardcoded, it should be defined in the CSS styles so that users can override it
  - I am torn w.r.t. hundreds of shades. An alternative would be to display just a couple of (tens ?) shades and the rest leave it at default color. This way it would be easier to identify changes to adjacent lines - say if some lines in the same neighborhood were changed by couple of recent changesets it would be hard to identify which lines were changed by which changeset since the shade difference would be negligible.

**Comment by vladak** [comment]
Any updates ?

**Comment by vladak** [comment]
Even if you are not going to make any changes, can you accept OCA pls ?

**Comment by vladak** [comment]
pushed as c904440d32e181d0920989f48a1c903c329b4b36

**Comment by dobrou** [comment]
No updates, unfortunately I didn't have much more free time to spend on this yet.
If somebody takes this as feature request including prototype, I could not have any complaints. Otherwise I will finish my work as soon I'll have some free evenings :)

Yes, my OCA was signed and delivered already.

**Comment by vladak** [comment]
Don't worry, it is now sitting in master, @tulinkry will give it some tuning.

**Pull request closed without merging**